### PR TITLE
Update akka to 1.4.2

### LINKED ISSUE #1455: Update Akka to up-to-date version
According to https://github.com/akkadotnet/akka.net/issues/2378#event-3071191410 there might be benefits in the P2P on the last version.

As #611 mentioned and discussed in other threads.

image

### PR DESCRIPTION
**Pull request by ShawnYun**
Closes #1455
As #1455 , we can update Akka to 1.4.2

### CODE REVIEW

**Comment by shargon** [comment]
We are making tests :)

**Comment by shargon** [comment]
You can check master vs #1466 here

http://neocharts.red4sec.com/viewer/?u=http://neocharts.red4sec.com/1583760675/report.json&v=http://neocharts.red4sec.com/1583762924/report.json&g=txBlock[low-tx-load|secondary;low-tx-load;txBlock]allblocks,txBlock[high-tx-load|secondary;high-tx-load;txBlock]allblocks,MemPool[high-tx-load|secondary;high-tx-load;MemPool]allblocks

With the current parameters (500 tx per block) there are not to many diference. We should check with higher values

**Comment by ShawnYun** [comment]
@shargon The  new Akka improves the ability and stability to handle a large number of messages. So the result of Tx/Block and Memory Pool has not to many diference. Maybe we should test the maximum number of messages the system can process per second. And the stability of the system under different conditions.

File: src/neo/neo.csproj [hunk 1]
```diff
@@ -21,7 +21,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Akka" Version="1.3.16" />
+    <PackageReference Include="Akka" Version="1.4.1-rc1" />
```
Comment (cloud8little) [inline src/neo/neo.csproj]: suggest that we should use a release version. this seems to be a release candidate one.

Comment (ShawnYun) [reply src/neo/neo.csproj]: @cloud8little Yes, There is no release version of akka 1.4. We can wait for the release version. I will test the new akka first.

**Comment by belane** [comment]
@ShawnYun it seems that the number of p2p messages and the transmitted bytes maintain a similar value with master.

http://neocharts.red4sec.com/viewer/?u=http://neocharts.red4sec.com/1583760675/report.json&v=http://neocharts.red4sec.com/1583762924/report.json&g=totBytes[high-tx-load|secondary;high-tx-load;totBytes]allblocks,messCount[high-tx-load|secondary;high-tx-load;messCount]allblocks
_(Green this PR / Black master)_

But although Consensus and Transactions work, the following error message appears repeatedly on all nodes.

```
node-2    | [WARNING][03/14/2020 18:20:33][Thread 0012][akka://NeoSystem/system/IO-TCP/$g] Received Write command before Register command. It will be buffered until Register will be received (buffered write size is 52 bytes)
node-1    | [WARNING][03/14/2020 18:20:33][Thread 0011][akka://NeoSystem/system/IO-TCP/$a/$d] Received Write command before Register command. It will be buffered until Register will be received (buffered write size is 52 bytes)
node-6    | [WARNING][03/14/2020 18:20:33][Thread 0004][akka://NeoSystem/system/IO-TCP/$a/$d] Received Write command before Register command. It will be buffered until Register will be received (buffered write size is 52 bytes)
node-5    | [WARNING][03/14/2020 18:20:34][Thread 0005][akka://NeoSystem/system/IO-TCP/$f] Received Write command before Register command. It will be buffered until Register will be received (buffered write size is 52 bytes)
node-1    | [WARNING][03/14/2020 18:20:34][Thread 0012][akka://NeoSystem/system/IO-TCP/$a/$f] Received Write command before Register command. It will be buffered until Register will be received (buffered write size is 52 bytes)
node-4    | [WARNING][03/14/2020 18:20:34][Thread 0013][akka://NeoSystem/system/IO-TCP/$a/$f] Received Write command before Register command. It will be buffered until Register will be received (buffered write size is 52 bytes)
node-3    | [WARNING][03/14/2020 18:20:34][Thread 0005][akka://NeoSystem/system/IO-TCP/$b] Received Write command before Register command. It will be buffered until Register will be received (buffered write size is 52 bytes)
node-4    | [WARNING][03/14/2020 18:20:40][Thread 0017][akka://NeoSystem/system/IO-TCP/$i] Received Write command before Register command. It will be buffered until Register will be received (buffered write size is 52 bytes)
node-5    | [WARNING][03/14/2020 18:20:44][Thread 0011][akka://NeoSystem/system/IO-TCP/$a/$f] Received Write command before Register command. It will be buffred until Register will be received (buffered write size is 52 bytes)
...
```

**Comment by Tommo-L** [comment]
We'll close this PR to wait for a better stable version. In our recently test, akka v1.3.16 seems better than v1.4.1 https://github.com/akkadotnet/akka.net/issues/4329#issuecomment-599381429

**Comment by ShawnYun** [comment]
@belane Thank you, I'll deal with this problem.

**Comment by ShawnYun** [comment]
@belane  According to Akka's official [answer](https://github.com/akkadotnet/akka.net/issues/4340), This is one of the logging system improvements we added into new release. This is marked as Warning (not Error) since we are just buffering such writes (unlike dropping them as it was before v1.4), and then flushing buffer once your Tcp.Register message is arrived. So basically all is working, no data loss occurs.

**Comment by ShawnYun** [comment]
Akka1.4.2 has been released. One change is "Reintroduce old HOCON, lay groundwork for gradual introduction of HOCON stand-alone". So there is no need to use the new hocon now.

**Comment by ShawnYun** [comment]
According to the test results, the speed and stability of Akka1.4.1 was better than 1.3.16.
微信图片编辑_20200317160412

**Pull request merged**